widget/embedded/src/utils/colors.ts

```diff
@@ -1,141 +1,185 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+// Types
 import type { WidgetColors, WidgetColorsKeys } from '../types';
 
-export function generateShades(
-  baseColor: string,
-  prefix: string,
-  reverse?: boolean
-): { [key: string]: string } {
-  const BASE_SHADE_INDEX = 100;
-  const NUMBER_OF_COLORS = 9;
-  const SHADING_INCREMENT_MAX = 10; // For bright colors
-  const SHADING_INCREMENT_MIN = 6; // For dark colors
-
-  const HALF_NUMBER_OF_COLORS = 4;
-
-  const shadeIncrements: number[] = [];
-  let increment = 90;
-  for (let i = 0; i < NUMBER_OF_COLORS; i++) {
-    shadeIncrements.push(increment);
-    if (i + 1 === HALF_NUMBER_OF_COLORS) {
-      increment = 0;
-    } else {
-      increment -=
-        i >= HALF_NUMBER_OF_COLORS
-          ? SHADING_INCREMENT_MIN
-          : SHADING_INCREMENT_MAX;
+import {
+  RANGO_MAIN_NEUTRAL_COLOR,
+  RANGO_NEUTRAL_COLORS,
+} from '../constants/colors';
+
+type RGB = {
+  red: number;
+  green: number;
+  blue: number;
+};
+
+type ShadeOrTintFunction = (rgb: RGB, step: number) => RGB;
+
+const HEX_BASE = 16;
+const MAX_COLOR_VALUE = 255;
+const PERCENT = 0.11;
+
+// Utility Functions
+
+// Determines if there are color changes between base and expanded colors.
+export function findChangedColors(
```

Comment:
```
/*
We only need to expand those colors that doesn't match with baseColors.
Our default theme colors don't follow our formula for generating colors and actually they are custom colors. By checking `baseColors` and `expandColors` we will remove those colors that are default colors to avoid generating colors for them. 
*/

function removeDefaultColorsFromExpand(...){
// ...
}
```

Reply:
Please rename the function and add the comment.